effects/sleep.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+import time
+
+import click
+
+from rpi_ws281x import PixelStrip, Color
+
+# LED strip default configuration:
+#LED_COUNT = 50       # Number of LED pixels.
+LED_PIN = 18          # GPIO pin connected to the pixels (18 uses PWM!).
+LED_FREQ_HZ = 800000  # LED signal frequency in hertz (usually 800khz)
+LED_DMA = 10          # DMA channel to use for generating signal (try 10)
+LED_BRIGHTNESS = 255  # Set to 0 for darkest and 255 for brightest
+LED_INVERT = False    # True to invert the signal (when using NPN transistor level shift)
+LED_CHANNEL = 0       # set to '1' for GPIOs 13, 19, 41, 45 or 53
+
+@click.command()
+@click.option('-l', '--leds', type=int, default=50, show_default=True, help='Amount of LEDs in the string')
+@click.option('-s', '--speed', type=float, default=1.0, show_default=True, help='Speed of the effect')
+@click.option('-c', '--color', type=str, default='FF0000', show_default=True, help='RGB Value for the effect. You need to provide this value as a hexadecimal.')
+def main(leds: int, speed: float, color: str):
+    '''
+    This is a sleep effect which provide a light for the night
+    '''
+    if not color.startswith('0x'):
+        color = '0x' + color
+        
+    print(f'Sleep effect started with following parameter: leds={leds}, speed={speed}, color={color}')
+    color_dict = parseColorString(color)
+    
+    strip = PixelStrip(leds, LED_PIN, LED_FREQ_HZ, LED_DMA, LED_INVERT, LED_BRIGHTNESS, LED_CHANNEL)
+    strip.begin() # Intialize the library (must be called once before other functions).
+    
+    runEffect(strip, speed, color_dict['red'], color_dict['green'], color_dict['blue']);
+    pass
+
+def parseColorString(color: str):
+    '''
+    Parse the color given as string to single RGB colors as int values and return the dictionary with
+    parsed values
+    '''
+    color_int_val = int(color, base=16)
+    
+    return {
+        'red': (color_int_val >> 16) & 0xFF,
+        'green': (color_int_val >> 8) & 0xFF,
+        'blue': color_int_val & 0xFF
+    }
+
+def runEffect(strip, speed, red_max_val, green_max_val, blue_max_val):
+    SLEEP_TIME = 0.005 / speed
```

Comment:
* The factor should be a constant.
* `SLEEP_TIME` should be written in lowercase, since it's rather a variable than a constant.

Reply:
> The factor should be a constant.
I just don´t know how to name this factor. And to name the constant only `FACTOR` bring no value from my perspective. The value itself was only adjusted by "experience" ... by looking at the effect and finding out the speed i like. The "speed" param was added, because for Jan it was too slow. But make just a suggestion to this line and we can talk about :)

> SLEEP_TIME should be written in lowercase, since it's rather a variable than a constant.
Yeah ... i wanted a const var here ... but didn´t know if i could do this ... but yes, i will adjust the var name

Reply:
1. My suggestions: remove `SLEEP_TIME` and just use `speed` instead. There is no value in having a magic constant IMHO.
* You will run the method with input `sleep=0.005`
* Jan will run the method with his own preferred input.

2. We had the discussion in another project:
A final variable is not a constant, IMHO. It is still a variable.

Reply:
1. There is a value to have additional configuration factor.

So imagine you have several effects and each has his own tempo .. so sleep has a need of a delay of 5ms (to have nice speed in eyes of the author of the effect), but other effects will need another sleep time or even multiple different values. So if we use just "speed" and you configure the effects via "nodeRED" for example you will have a need to have a look on each effect code every time you configure an effect with a different speed. For effects with multiple sleep timers, you will have a need to configure multiple values keeping its difference proportional to each other.

Additional speed values make it just simple and because it defines a default speed at value 1.0 you will exactly know (just looking on your parameter) how far you changed the speed in comparison to the original value.

2. Yes we had an discussion and we did not end the discussion :P
Yes, final variables are constants, but we write only "static final" vars in upper case. I don' t know if we can call static final's a variables, because at the compile time they will be replaced by its value and after compilation is done, the don't exist anymore (as i know).

But anyway ... python is not java and python don't have constants, but you can indicate a variable as an constant by writing it in upper case. But, according to PEP8's rules on [constants](https://peps.python.org/pep-0008/#constants)
> Constants are usually defined on a module level and written in all capital letters with underscores separating words. Examples include MAX_OVERFLOW and TOTAL.

So i would also apply this rules to this variable :D

Reply:
Except that `SLEEP_TIME` changes depending on input, whereas `MAX_OVERFLOW` and `TOTAL` don't.